docs/user/subprojects.rst

```diff
@@ -39,6 +39,18 @@ If the example project ``Foo`` was set up with a custom domain,
 respectively be at: https://docs.example.com/en/latest/ and
 https://docs.example.com/projects/bar/en/latest/
 
+Custom domain on subprojects
+----------------------------
+
+Adding a custom domain to a subproject is allowed,
+but your documentation will always be served from
+the domain of the parent project.
```

Comment:
Instead, perhaps we just lie and say "This is not supported" :smile: 

The best UX would probably be to disallow creation of new domains entirely in this scenario. The lighter version of this could be to start with this and a warning though -- but i suppose disabling the button if the project is a subproject is a harmless addition too.

Reply:
Opened https://github.com/readthedocs/readthedocs.org/issues/8985 to discuss that option